src/System.IO/src/System/IO/FileStreamHelpers.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+
+namespace System.IO
+{
+    // Use reflection to access the implementation of FileStream in System.IO.FileSystem.dll. While far from ideal,
+    // we do this to avoid having to directly reference FileSystem as doing so introduces a dependency cycle. A more
+    // permanent solution would be to merge the IO and FileSystem assemblies.
+    internal static class FileStreamHelpers
+    {
+        private const int FileMode_Create = 2;
+        private const int FileMode_Open = 3;
+        private const int FileMode_Append = 6;
+        private const int FileAccess_Read = 1;
+        private const int FileAccess_Write = 2;
+
+        private static MethodInfo s_fileOpen;
+
+        public static Stream CreateFileStream(string path, bool write, bool append)
+        {
+            if (s_fileOpen == null)
+            {
+                s_fileOpen = GetFileOpenFunction();
+            }
+            int filemode = !write ? FileMode_Open : append ? FileMode_Append : FileMode_Create;
+            int fileaccess = write ? FileAccess_Write : FileAccess_Read;
+            object[] parameters = new object[] { path, filemode, fileaccess };
+            return s_fileOpen?.Invoke(null, parameters) as Stream;
```

Comment:
You'd need to measure, but I think it'll be measurably faster to create a delegate for the MethodInfo in GetFileOpenFunction using MethodInfo.CreateDelegate, store that delegate in the static field, and then just invoke that delegate here as if you were calling a function (if nothing else, it'll avoid needing to manually allocate the array).  I realize that's not how it's done in the Process-related code you copied; we should probably change that there, too.  I also realize this is only meant to be a temporary hack, but just in case :)